src/scripts/main.js

```diff
@@ -1,11 +1,26 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    const handleEvent = (e) => {
+      element.removeEventListener(eventName, handleEvent);
+
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    };
+
+    element.addEventListener(eventName, handleEvent);
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const div = document.createElement('div');
+
+  div.className = 'message';
+  div.textContent = message;
+
+  document.body.append(div);
 };
```

Comment:
The function `printMessage` is tightly coupled with the document body. This means that you can't use this function to append a message to any other element. Consider passing the parent element as an argument to the function. This will make your function more flexible and reusable.